src/main/java/WayofTime/bloodmagic/api/registry/AltarRecipeRegistry.java

```diff
@@ -82,17 +88,46 @@ public AltarRecipe(ItemStack input, ItemStack output, EnumAltarTier minTier, int
             this(input, output, minTier, syphon, consumeRate, drainRate, false);
         }
 
+        /**
+         * {@link OreDictionary} version of the above method
+         */
+        public AltarRecipe(String oreDictInput, ItemStack output, EnumAltarTier minTier, int syphon, int consumeRate, int drainRate)
+        {
+            this(OreDictionary.getOres(oreDictInput), output, minTier, syphon, consumeRate, drainRate, false);
```

Comment:
You don't have to use `this`. Handle the fields manually. The check *needs* to be done here.